core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/pythonUDF/ArrowUtils.scala

```diff
@@ -0,0 +1,242 @@
+        case _: ArrowType.Int =>
+          vector.getField.getFieldType.getType.asInstanceOf[ArrowType.Int].getBitWidth match {
+            case 16 | 32 =>
+              vector
+                .asInstanceOf[IntVector]
+                .set(index, !isNull, if (isNull) 0 else value.asInstanceOf[Int])
+
+            case 64 | _ =>
+              vector
+                .asInstanceOf[BigIntVector]
+                .set(index, !isNull, if (isNull) 0 else value.asInstanceOf[Long])
+          }
+
+        case _: ArrowType.Bool =>
+          vector
+            .asInstanceOf[BitVector]
+            .set(index, !isNull, if (isNull) 0 else value.asInstanceOf[Boolean])
+
+        case _: ArrowType.FloatingPoint =>
+          vector
+            .asInstanceOf[Float8Vector]
+            .set(index, !isNull, if (isNull) 0 else value.asInstanceOf[Double])
+
+        case _: ArrowType.Timestamp =>
+          vector
+            .asInstanceOf[TimeStampVector]
+            .set(
+              index,
+              !isNull,
+              if (isNull) 0L
+              else AttributeTypeUtils.parseField(value, AttributeType.LONG).asInstanceOf[Long]
+            )
+
+        case _: ArrowType.Utf8 =>
+          if (isNull) vector.asInstanceOf[VarCharVector].setNull(index)
+          else
+            vector
+              .asInstanceOf[VarCharVector]
+              .set(index, value.asInstanceOf[String].getBytes(StandardCharsets.UTF_8))
+
+      }
+    }
+
+    vectorSchemaRoot.setRowCount(vectorSchemaRoot.getRowCount + 1)
+  }
+
+  /**
+    * Converts an Amber schema into Arrow schema.
+    *
+    * @param schema The Texera Schema.
+    * @return An Arrow Schema.
+    */
+  def fromTexeraSchema(schema: Schema): org.apache.arrow.vector.types.pojo.Schema = {
+    val arrowFields = new util.ArrayList[Field]
+
+    for (amberAttribute <- schema.getAttributes) {
+      val name = amberAttribute.getName
+      val field = Field.nullablePrimitive(name, fromAttributeType(amberAttribute.getType))
+      arrowFields.add(field)
+    }
+    new org.apache.arrow.vector.types.pojo.Schema(arrowFields)
+  }
+
+  /**
+    * Converts an AttributeType into an ArrowType (PrimitiveType).
+    *
+    * @param srcType The AttributeType to be converted.
+    * @throws AttributeTypeException if the type cannot be converted.
+    * @return A PrimitiveType, a type of ArrowType, does not handle complex data.
+    */
+  @throws[AttributeTypeException]
+  def fromAttributeType(srcType: AttributeType): PrimitiveType = {
+    srcType match {
+      case AttributeType.INTEGER =>
+        new ArrowType.Int(32, true)
+
+      case AttributeType.LONG =>
+        new ArrowType.Int(64, true)
+
+      case AttributeType.DOUBLE =>
+        new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)
+
+      case AttributeType.BOOLEAN =>
+        ArrowType.Bool.INSTANCE
+
+      case AttributeType.TIMESTAMP =>
+        new ArrowType.Timestamp(NANOSECOND, "UTC")
```

Comment:
It seems arrow type is in nano seconds, but the `parseLong` in `AttributeTypeUtils` returns miliseconds

Reply:
changed to `MILLISECOND`